app/controllers/teams/users_controller.rb

```diff
@@ -1,5 +1,33 @@
 class Teams::UsersController < AuthenticatedController
-   before_action -> { authorize team, policy_class: TeamUsersPolicy }
+    before_action -> { authorize team, policy_class: TeamUsersPolicy }
+
+    def new
+      @user = User.new
+    end
+  
+    def create
+      add_email = user_params.require(:email).downcase
+      if not valid_email_address?(add_email)
+        @user = User.new
+        flash[:error] = I18n.t('teams.users.create.invalid_email', email: add_email)
+        render :new
```

Comment:
I'd consider moving all of these `render :new` in each fail to the top level of this method, and add a return on the success path.